### PR TITLE
Fix the MultiValueMetricsAggregator and reduce overhead

### PR DESCRIPTION
**Pull request by LuQQiu**
Fix the bug of `MetricsSystem.getMasterMetrics()`. It previously only returns the metric stored in MetricsRegistry with the exact same name. In this PR, it returns the metrics that start with the given metric name (the metrics can have multiple tags).

Reduce the logics in update `MultiValueMetricsAggregator` values, reduce the traverse of the whole metrics.

### CODE REVIEW

**Comment by LuQQiu** [comment]
Tests in error: 
alluxio.server.ft.journal.ufs.UfsJournalIntegrationTest.file(alluxio.server.ft.journal.ufs.UfsJournalIntegrationTest)
  Run 1: UfsJournalIntegrationTest.file:328->fileTestUtil:332->createFsMasterFromJournal:603 » Runtime
  Run 2: UfsJournalIntegrationTest.file:328->fileTestUtil:332->createFsMasterFromJournal:603 » Runtime

**Comment by LuQQiu** [comment]
Jenkins, test this please

File: core/common/src/main/java/alluxio/metrics/MetricsSystem.java [hunk 1]
```diff
@@ -621,19 +625,25 @@ public static Timer timer(String name) {
...
-    Set<Metric> set = new HashSet<>();
-    String fullName = getMasterMetricName(name);
-    Metric alluxioMetric = getMetricValue(fullName);
-    if (alluxioMetric != null) {
-      set.add(alluxioMetric);
+  public static Map<String, Set<Metric>> getMasterMetrics(Set<String> metricNames) {
+    Map<String, Set<Metric>> res = new HashMap<>();
+    for (Map.Entry<String, com.codahale.metrics.Metric> entry
+        : METRIC_REGISTRY.getMetrics().entrySet()) {
+      Matcher matcher = METRIC_NAME_PATTERN.matcher(entry.getKey());
```
[the remark this answers is not in the document]
Comment (LuQQiu) [reply core/common/src/main/java/alluxio/metrics/MetricsSystem.java]: This patten helps reduce the traverse of the given name set, the complexity is O(n), n = metrics number in the whole metrics system.
If using startsWtih, the complexity is O(n*m), m = number of elements in the given metrics name set.
That's why I changed from `startsWith` to pattern matching.

**Comment by LuQQiu** [comment]
Tests in error: 
alluxio.server.ft.journal.MultiMasterJournalTest.testCheckpointReplay(alluxio.server.ft.journal.MultiMasterJournalTest)
  Run 1: MultiMasterJournalTest.before:53 » IO java.util.concurrent.TimeoutException: T...
  Run 2: MultiMasterJournalTest.before:53 » IO java.util.concurrent.TimeoutException: T...

alluxio.server.ft.journal.ufs.UfsJournalIntegrationTest.setAcl(alluxio.server.ft.journal.ufs.UfsJournalIntegrationTest)
  Run 1: UfsJournalIntegrationTest.setAcl:590->aclTestUtil:594->createFsMasterFromJournal:603 » Runtime
  Run 2: UfsJournalIntegrationTest.setAcl:590->aclTestUtil:594->createFsMasterFromJournal:603 » Runtime

**Comment by LuQQiu** [comment]
Jenkins, test this please

File: core/server/master/src/main/java/alluxio/master/metrics/DefaultMetricsMaster.java [hunk 2]
```diff
@@ -53,8 +51,10 @@
  */
 public class DefaultMetricsMaster extends CoreMaster implements MetricsMaster, NoopJournaled {
   private static final Logger LOG = LoggerFactory.getLogger(DefaultMetricsMaster.class);
-  private final Set<MultiValueMetricsAggregator> mMultiValueMetricsAggregatorRegistry =
-      new HashSet<>();
+  // A map from the to be aggregated metric name to aggregator itself
+  // This registry only holds aggregator for master metrics
+  private final Map<String, MultiValueMetricsAggregator>
+      mAggregatorRegistry = new HashMap<>();
```
Comment (ZacBlanco) [inline core/server/master/src/main/java/alluxio/master/metrics/DefaultMetricsMaster.java]: nit: This seems like an odd line break. Does it not fit on one line?

**Comment by ZacBlanco** [comment]
One minor nit, otherwise LGTM

File: core/server/master/src/main/java/alluxio/master/metrics/DefaultMetricsMaster.java (hunk 2 above)
Comment (LuQQiu) [reply core/server/master/src/main/java/alluxio/master/metrics/DefaultMetricsMaster.java]: Fixed, thanks

**Comment by LuQQiu** [comment]
alluxio-bot,  merge this please

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**